system-addon/lib/TopSitesFeed.jsm

```diff
@@ -210,23 +246,41 @@ this.TopSitesFeed = class TopSitesFeed {
 
   /**
    * Pin a site at a specific position saving only the desired keys.
+   * @param customScreenshotURL {string} User set URL of preview image for site
+   * @param label {string} User set string of custom site name
    */
-  _pinSiteAt({label, url}, index) {
+  async _pinSiteAt({customScreenshotURL, label, url}, index) {
     const toPin = {url};
     if (label) {
       toPin.label = label;
     }
+    if (customScreenshotURL) {
+      toPin.customScreenshotURL = customScreenshotURL;
+    }
     NewTabUtils.pinnedLinks.pin(toPin, index);
+
+    await this._updateLinkCustomScreenshot({customScreenshotURL, url});
```

Comment:
Why not just pass in `toPin`?

Reply:
we need to pass the explicit `null` value of `customScreenshotURL`